docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -25,308 +29,444 @@ function su_humsci_profile_final_task() {
+              'type' => $storage->getType(),
+              'field_name' => $field_name,
+              'entity_type' => 'hs_entity',
+              'settings' => $storage->getSettings(),
+              'cardinality' => $storage->getCardinality(),
+            ])->save();
+          }
+
+          // Create the new field config on the hs entity to match the ECK.
+          if (!FieldConfig::loadByName('hs_entity_type', $new_bundle_id, $field_name)) {
+            FieldConfig::create([
+              'uuid' => _su_humsci_profile_get_uuid("field.field.hs_entity.$new_bundle_id.$field_name"),
+              'field_type' => $config->getType(),
+              'entity_type' => 'hs_entity',
+              'bundle' => $new_bundle_id,
+              'field_name' => $field_name,
+              'label' => $config->label(),
+              'settings' => $config->getSettings(),
+            ])->save();
+
+            // Copy the form and display settings for the field.
+            if ($form_settings = $eck_form->getComponent($field_name)) {
+              $hs_entity_form->setComponent($field_name, $form_settings);
+            }
+
+            if ($display_settings = $eck_display->getComponent($field_name)) {
+              $hs_entity_display->setComponent($field_name, $display_settings);
+            }
+          }
+        }
+      }
+
+      // Copy the third party settings to capture DS or field group settings.
+      $hs_entity_form->set('third_party_settings', $eck_form->get('third_party_settings'));
+      $hs_entity_display->set('third_party_settings', $eck_display->get('third_party_settings'));
+
+      $hs_entity_form->save();
+      $hs_entity_display->save();
+    }
+  }
 }
 
 /**
- * Disable preview mode on all content types.
+ * Migrate ECKs to the new types.
  */
-function su_humsci_profile_update_8203() {
-  $node_types = \Drupal::entityTypeManager()
-    ->getStorage('node_type')
-    ->loadMultiple();
-  $config_factory = \Drupal::configFactory();
-  foreach (array_keys($node_types) as $type) {
-    $config_factory->getEditable("node.type.$type")
-      ->set('preview_mode', 0)
-      ->save();
+function su_humsci_profile_update_9701(&$sandbox) {
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $eck_info = _su_humsci_profile_update_eck_info();
+  if (!isset($sandbox['count'])) {
+    $sandbox['items'] = [];
+    $sandbox['convert'] = [];
+
+    // Store all the ECK ids into the sandbox for batch processing.
+    foreach (array_keys($eck_info) as $eck_name) {
+      $ids = $entity_type_manager->getStorage($eck_name)
+        ->getQuery()
+        ->accessCheck(FALSE)
+        ->execute();
+      foreach ($ids as $id) {
+        $sandbox['items'][] = "$eck_name:$id";
+      }
+    }
+    $sandbox['items'] = array_filter($sandbox['items']);
+    $sandbox['count'] = count($sandbox['items']);
+    \Drupal::messenger()
+      ->addStatus(t('Updating @count ECK items.', ['@count' => count($sandbox['items'])]));
```

Comment:
This is pedantic, but should the messenger message say "Migrating" instead of "Updating" 😆

Reply:
hmm. depends on which way you look at it I guess. Or should it be `UpGRADEing` them? haha